### ISSUE TITLE
Session autosave is unreliable

### ISSUE THREAD

**Issue by jknoos** [qutebrowser/qutebrowser#8219]
**Version info**:

qutebrowser v3.1.0
Git commit: 
Backend: QtWebEngine 6.7.1, based on Chromium 118.0.5993.220 (from api)
Qt: 6.7.1

**Does the bug happen if you start with `--temp-basedir`?**:

Yes.

**Description**

The age of `~/.local/share/qutebrowser/sessions/_autosave.yml` can exceed one minute by a large margin.

```
[~]$ echo $(($(date +%s) - $(date +%s -r ~/.local/share/qutebrowser/sessions/_autosave.yml))) seconds
423 seconds
```

Tabs were closed in the interim, but no new tabs opened.

As I'm raising this bug I realize I've either misunderstood the purpose of this file or there is a discrepancy between the documentation and the implementation as the latter only saves the file when page loads are completed, not when tabs are closed.

**How to reproduce**

1. Start qutebrower.
2. Open some tabs, wait one minute.
3. Close some tabs. Observe `_autosave.yml` not getting updated.

**Comment by The-Compiler** [comment]
Good catch - `sessions.session_manager.save_autosave()` indeed is only called in `AbstractTab._on_load_finished()`. It should probably be called on every major event that affects session state, and I agree that closing a tab definitely is one of those.

**Comment by neeasade** [comment]
FWIW I've had `d` bound to save the session for a few years now:

``` py
d_sync = ';;'.join([
    'tab-close',
    'set messages.timeout 1',
    'session-save --force _autosave',
    'set messages.timeout 2000',
    ])


config.bind('d', d_sync, mode='normal')
```
